src/mapml/handlers/QueryHandler.js

```diff
@@ -104,74 +104,72 @@ export var QueryHandler = L.Handler.extend({
+        }).then((mapml) => {
+          if (contenttype.startsWith("text/mapml")) {
+            return handleMapMLResponse(mapml, e.latlng);
+          } else {
+            return handleOtherResponse(mapml, layer, e.latlng);
+          }
+        }).catch((err) => {
+          console.log('Looks like there was a problem. Status: ' + err.message);
+        });
       }
-      function handleMapMLResponse(response, loc) {
-          return response.text().then(mapml => {
-              // bind the deprojection function of the layer's crs 
-              var _unproject = L.bind(crs.unproject, crs);
-              function _coordsToLatLng(coords) {
-                  return _unproject(L.point(coords));
-              }
-              var parser = new DOMParser(),
-                  mapmldoc = parser.parseFromString(mapml, "application/xml"),
-                  f = M.mapMlFeatures(mapmldoc, {
-                  // pass the vector layer a renderer of its own, otherwise leaflet
-                  // puts everything into the overlayPane
-                  renderer: L.svg(),
-                  // pass the vector layer the container for the parent into which
-                  // it will append its own container for rendering into
-                  pane: container,
-                  color: 'yellow',
-                  // instead of unprojecting and then projecting and scaling,
-                  // a much smarter approach would be to scale at the current
-                  // zoom
-                  coordsToLatLng: _coordsToLatLng,
-                  imagePath: M.detectImagePath(map.getContainer())
-              });
-              f.addTo(map);
+      function handleMapMLResponse(mapml, loc) {
+        var parser = new DOMParser(),
+            mapmldoc = parser.parseFromString(mapml, "application/xml"),
+            f = M.mapMlFeatures(mapmldoc, {
+            // pass the vector layer a renderer of its own, otherwise leaflet
+            // puts everything into the overlayPane
+            renderer: L.svg(),
+            // pass the vector layer the container for the parent into which
+            // it will append its own container for rendering into
+            pane: container,
+            //color: 'yellow',
+            // instead of unprojecting and then projecting and scaling,
+            // a much smarter approach would be to scale at the current
+            // zoom
+            //coordsToLatLng: _coordsToLatLng,
+            _leafletLayer: layer,
+            imagePath: M.detectImagePath(map.getContainer()),
+            query: true,
+        });
+        f.addTo(map);
 
-              var c = document.createElement('iframe');
-              c.csp = "script-src 'none'";
-              c.style = "border: none";
-              c.srcdoc = mapmldoc.querySelector('feature properties').innerHTML;
+        let div = L.DomUtil.create("div", "mapml-popup-content"),
+            c = L.DomUtil.create("iframe");
+        c.csp = "script-src 'none'";
```

Comment:
I don't think there's a need to set the  `csp` attribute anymore, I suggest removing this line.